### ISSUE TITLE
New keyboard shortcut: add a new line on the previous line

### ISSUE THREAD

**Issue by exalted** [zed-industries/community#4]
This shortcut should "push down" lines following the cursor and insert new lines where the cursor currently is.

It's `Cmd-shift-enter` in vscode.

**Comment by rkusa** [comment]
I'd also expect the opposite to exist: `CMD+Enter` to insert a new line below the current line (and move the cursor to it). This shortcut exists in VSCode and Sublime (and others too I'd assume). The benefit of `CMD+Enter` compared to a simple `Enter` is that you are not splitting the current line even if your cursor is not at the end of the current line.

In VIM terms the shortcuts correspond to:
- `CMD+Enter`: `o`
- `CMD+Shift+Enter`: `O`

**Comment by rkusa** [comment]
I was able to add both shortcuts by adding the following to my `keymap.json`:

```json
[
    {
        "context": "Editor",
        "bindings": {
            "cmd-enter": "vim::InsertLineBelow",
            "cmd-shift-enter": "vim::InsertLineAbove"
        }
    }
]
```

It would still be great if those keybindings would exist by default.

**Comment by petros** [comment]
This is merged and should ship soon.

**Comment by exalted** [comment]
W00t!!! 🚀

**Comment by hovsater** [comment]
Shipped in [0.83.1](https://zed.dev/releases/0.83.1). 🎉
